tests/unit/test_charm.py

```diff
@@ -1,74 +1,118 @@
-        self.assertEqual(self.harness.model.unit.status, ActiveStatus())
-
-    def test_config_changed_valid_cannot_connect(self):
-        # Trigger a config-changed event with an updated value
-        self.harness.update_config({"log-level": "debug"})
-        # Check the charm is in WaitingStatus
-        self.assertIsInstance(self.harness.model.unit.status, WaitingStatus)
-
-    def test_config_changed_invalid(self):
-        # Ensure the simulated Pebble API is reachable
-        self.harness.set_can_connect("httpbin", True)
-        # Trigger a config-changed event with an updated value
-        self.harness.update_config({"log-level": "foobar"})
-        # Check the charm is in BlockedStatus
-        self.assertIsInstance(self.harness.model.unit.status, BlockedStatus)
+            )
+        },
+    )
+    return relation_id
+
+
+def test_not_leader(harness):
+    """Test with unit not being leader."""
+    harness.set_leader(False)
+
+    harness.charm.on.login_ui_pebble_ready.emit(CONTAINER_NAME)
+
+    assert (
+        "status_set",
+        "waiting",
+        "Waiting to connect to Login_UI container",
+        {"is_app": False},
+    ) in harness._get_backend_calls()
+
+
+def test_install_can_connect(harness):
+    """Test installation with connection."""
+    harness.set_can_connect(CONTAINER_NAME, True)
+    harness.charm.on.login_ui_pebble_ready.emit(CONTAINER_NAME)
+
+    assert harness.charm.unit.status == ActiveStatus()
+
+
+def test_install_can_not_connect(harness):
+    """Test installation with connection."""
+    harness.set_can_connect(CONTAINER_NAME, False)
+    harness.charm.on.login_ui_pebble_ready.emit(CONTAINER_NAME)
+
+    assert harness.charm.unit.status == WaitingStatus(
+        "Waiting to connect to Login_UI container"
+    )
+
+
+def test_layer_updated(harness) -> None:
+    """Test Pebble Layers after updates."""
+    harness.set_can_connect(CONTAINER_NAME, True)
+    harness.charm.on.login_ui_pebble_ready.emit(CONTAINER_NAME)
+
+    harness.update_config({"hydra_url": TEST_HYDRA_URL})
+    harness.update_config({"kratos_url": TEST_KRATOS_URL})
+
+    expected_layer = {
+        "summary": "login_ui layer",
+        "description": "pebble config layer for identity platform login ui",
+        "services": {
+            CONTAINER_NAME: {
+                "override": "replace",
+                "summary": "identity platform login ui",
+                "command": "/id/identity_platform_login_ui",
+                "startup": "enabled",
+                "environment": {
+                    "HYDRA_ADMIN_URL": TEST_HYDRA_URL,
+                    "KRATOS_PUBLIC_URL": TEST_KRATOS_URL,
+                    "PORT": TEST_PORT,
+                },
+            }
+        },
+    }
+
+    assert harness.charm._login_ui_layer.to_dict() == expected_layer
+
+
+# finish test
+def test_fetch_endpoint_with_ingress_relation_data(harness) -> None:
```

Comment:
Don't write tests for private methods, only public interfaces. In the case of charms those are typically event hooks

Reply:
test has been deleted